text/0000-otlp-trace-data-format.md

```diff
@@ -0,0 +1,383 @@
+    int32 dropped_timed_events_count = 2;
+  }
+
+  // The included timed events.
+  TimedEvents timed_events = 11;
+
+  // A pointer from the current span to another span in the same trace or in a
+  // different trace. For example, this can be used in batching operations,
+  // where a single batch handler processes multiple requests from different
+  // traces or when the handler receives a request from a different project.
+  message Link {
+    // A unique identifier of a trace that this linked span is part of. The ID is a
+    // 16-byte array.
+    bytes trace_id = 1;
+
+    // A unique identifier for the linked span. The ID is an 8-byte array.
+    bytes span_id = 2;
+
+    // The relationship of the current span relative to the linked span: child,
+    // parent, or unspecified.
+    enum Type {
+      // The relationship of the two spans is unknown, or known but other
+      // than parent-child.
+      TYPE_UNSPECIFIED = 0;
+      // The linked span is a child of the current span.
+      CHILD_LINKED_SPAN = 1;
+      // The linked span is a parent of the current span.
+      PARENT_LINKED_SPAN = 2;
+    }
+
+    // The relationship of the current span relative to the linked span.
+    Type type = 3;
+
+    // A set of attributes on the link.
+    repeated AttributeKeyValue attributes = 4;
+
+    // The Tracestate associated with the link.
+    Tracestate tracestate = 5;
+  }
+
+  // A collection of links, which are references from this span to a span
+  // in the same or different trace.
+  message Links {
+    // A collection of links.
+    repeated Link link = 1;
+
+    // The number of dropped links after the maximum size was enforced. If
+    // this value is 0, then no links were dropped.
+    int32 dropped_links_count = 2;
+  }
+
+  // The included links.
+  Links links = 12;
+
+  // An optional final status for this span. Semantically when Status
+  // wasn't set it is means span ended without errors and assume
+  // Status.Ok (code = 0).
+  Status status = 13;
+
+  // An optional number of child spans that were generated while this span
+  // was active. If set, allows an implementation to detect missing child spans.
+  google.protobuf.UInt32Value child_span_count = 14;
+}
+
+// The `Status` type defines a logical error model that is suitable for different
+// programming environments, including REST APIs and RPC APIs. This proto's fields
+// are a subset of those of
+// [google.rpc.Status](https://github.com/googleapis/googleapis/blob/master/google/rpc/status.proto),
+// which is used by [gRPC](https://github.com/grpc).
+message Status {
+  // The status code. This is optional field. It is safe to assume 0 (OK)
+  // when not set.
+  int32 code = 1;
+
+  // A developer-facing error message, which should be in English.
+  string message = 2;
+}
+```
+
+### AttributeKeyValue
+
+```
+message AttributeKeyValue {
+  enum ValueType {
+    STRING  = 0;
+    BOOL    = 1;
+    INT64   = 2;
+    FLOAT64 = 3;
+    BINARY  = 4;
```

Comment:
What is binary value?

Reply:
It allows to store any data that does not fit the 4 other data types. Jaeger has this and I think it can be useful to allow carrying arbitrary payloads.

Reply:
This will be an API change and you need a separate RFC to add that support if you want. Carrying arbitrary payloads is not useful because you cannot display or understand them.

If you succeed to add the binary in the API as an attribute value I am ok with adding that, but this does not reflect the current API and is impossible to have this as a value. So please for the moment remove this.

Reply:
There is one use case when even if it is not in API it can still be useful. For OpenTelemetry Collector that receives Jaeger and exports OpenTelemetry this will be a useful addition because it preserves Jaeger's data type during translation. What do you think, is it worth having this in the protocol for this use case despite not being part of the API?